python/tvm/contrib/msc/core/transform/pattern.py

```diff
@@ -26,6 +27,27 @@
 from tvm.relax.transform import PatternCheckContext
 from tvm.relax.backend.pattern_registry import register_patterns
 from tvm.relay.op.contrib.register import register_pattern_table
+from tvm.contrib.msc.core.utils.namespace import MSCMap, MSCKey
+from tvm.contrib.msc.core import utils as msc_utils
+
+
+def msc_attrs_getter(context: PatternCheckContext, anchor: str = "out") -> Dict[str, str]:
+    """Get attributes for fused pattern
+
+    Returns
+    -------
+    attrs: dict<str,str>
+        The extra attributes for msc.
+    """
+
+    fused_cnt = MSCMap.get(MSCKey.FUSED_CNT, 0)
+    unique_name = "msc_fused_" + str(fused_cnt)
+    if anchor in context.annotated_expr:
```

Comment:
So what you really need from `PatternCheckContext` is just `annotated_expr`. Please follow my comment in https://github.com/apache/tvm/pull/16128#discussion_r1396423174. The getter function can just ingest `Map<String, Expr>` rather than `PatternCheckContext

Reply:
yes, the thing I need is the annotated_expr. I'll check your comment